pkgs/servers/mpd/default.nix

```diff
@@ -128,16 +136,15 @@ let
         #    Run-time dependency GTest found: YES 1.10.0
         gtest
       ]
-        ++ (lib.concatLists (lib.attrVals features_ featureDependencies))
+        ++ concatAttrVals features_ featureDependencies
         ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.AudioToolbox darwin.apple_sdk.frameworks.AudioUnit ];
 
       nativeBuildInputs = [
         meson
         ninja
         pkg-config
-        python3Packages.sphinx
-        doxygen
-      ];
+      ]
+        ++ concatAttrVals features_ nativeFeatureDependencies;
```

Comment:
If we enable docs by default, and we don't have [structuredAttrs](https://github.com/NixOS/nixpkgs/pull/72074) yet, I think we should also put them unconditionally in the nativeBuildInputs, and remove the whole logic behind it.

Reply:
The reason is https://github.com/NixOS/nixpkgs/pull/87940#issuecomment-629645759

Reply:
I created the `featureDependencies` attrset and the related functions in order to make the derivation easier to maintain. Structuring it this way obviates how dependencies are related to features, and makes it easier to spot mistakes. It was only turned into a customization mechanism because of a review request.
I'd be interested if anyone actually uses `mpdWithFeatures`, or `mpd-small` for that matter.

Reply:
I personally might have used mpdWithFeatures, if I had the time to setup my dream Nix based sound system :). Currently ARM support (for my Raspberry Pi) is still unsatisfying in Nixpkgs, unfortunately, so I don't use it (I'm using Debian (:sob:) on [BalenaOS](https://balena.io/)).

I'm personally "not afraid" of advanced Nix functions and alike when derivations are composed, I'm only trying to echo review comments I had in the past. Hence I won't make this issue a blocker for the update. Thanks for doing it anyway :).